src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/transformer/ApplicationsTransformerTest.kt

```diff
@@ -110,14 +115,29 @@ class ApplicationsTransformerTest {
   fun `transformJpaToApi transforms an Approved Premises application correctly`(args: Pair<ApiApprovedPremisesApplicationStatus, ApprovedPremisesApplicationStatus>) {
     val (apiStatus, jpaStatus) = args
 
-    val application = approvedPremisesApplicationFactory.withStatus(jpaStatus).withApArea(null).produce()
+    val applicantUserDetails = Cas1ApplicationUserDetailsEntityFactory().produce()
+    val caseManagerUserDetails = Cas1ApplicationUserDetailsEntityFactory().produce()
+
+    val application = approvedPremisesApplicationFactory
+      .withStatus(jpaStatus)
+      .withApArea(null)
+      .withApplicantUserDetails(applicantUserDetails)
+      .withApplicantIsCaseManager(true)
+      .withCaseManagerUserDetails(caseManagerUserDetails)
+      .produce()
+
+    every { mockCas1ApplicationUserDetailsTransformer.transformJpaToApi(applicantUserDetails) } returns Cas1ApplicationUserDetails("applicant", "", "")
+    every { mockCas1ApplicationUserDetailsTransformer.transformJpaToApi(caseManagerUserDetails) } returns Cas1ApplicationUserDetails("caseManager", "", "")
 
     val result = applicationsTransformer.transformJpaToApi(application, mockk()) as ApprovedPremisesApplication
 
     assertThat(result.id).isEqualTo(application.id)
     assertThat(result.createdByUserId).isEqualTo(user.id)
     assertThat(result.status).isEqualTo(apiStatus)
     assertThat(result.apArea).isNull()
+    assertThat(result.applicantUserDetails!!.name).isEqualTo("applicant")
+    assertThat(result.applicantIsCaseManager).isTrue()
+    assertThat(result.caseManagerUserDetails!!.name).isEqualTo("caseManager")
```

Comment:
This is a good point and the naming doesn't help. As implemented, caseManagerUserDetails is only populated if not the same as the applicant. I agree they should refer to the same entity given the current naming, but i think this may cause some headaches on the UI side. Do you think it would be acceptable if we could rename caseManagerUserDetails to something else ... although i'm struggling to think of what that could be

Reply:
altCaseManagerDetails? otherCaseManagerDetails?

It's not pretty but a little bit of javadoc on the properties may at least help disambiguate this approach